streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/StreamTableJoinNode.java

```diff
@@ -64,6 +70,16 @@ public void writeToTopology(final InternalTopologyBuilder topologyBuilder) {
         // Steam - KTable join only
         if (otherJoinSideNodeName != null) {
             topologyBuilder.connectProcessorAndStateStores(processorName, storeNames);
+            if (gracePeriod != null) {
+                for (final String storeName : storeNames) {
+                    if (!topologyBuilder.isStoreVersioned(storeName)) {
+                        throw new IllegalArgumentException("KTable must be versioned to use a grace period in a stream table join.");
+                    }
+                    if (gracePeriod.toMillis() > topologyBuilder.getHistoryRetention(storeName)) {
+                        throw new IllegalArgumentException("History history retention must be at least grace period, but it should be larger.");
```

Comment:
```suggestion
                        throw new IllegalArgumentException("History retention must be at least grace period.");
```
Must history retention be larger than or is it enough to be equal to the grace period? That is not clear from your original error message.

Reply:
This change is fine, just equal is enough, if not recommended